master/internal/user/postgres_users_intg_test.go

```diff
@@ -412,12 +424,156 @@ func addTestSession() (model.UserID, model.SessionID, string, error) {
 	return user.ID, session.ID, token, nil
 }
 
-func addTestUser(aug *model.AgentUserGroup) (*model.User, error) {
-	var user model.User
-	uid, err := Add(context.TODO(), &model.User{Username: uuid.NewString()}, aug)
+type opts func(*model.User)
+
+func addTestUser(aug *model.AgentUserGroup, opts ...opts) (*model.User, error) {
+	user := model.User{Username: uuid.NewString()}
+	for _, opt := range opts {
+		opt(&user)
+	}
+
+	uid, err := Add(context.TODO(), &user, aug)
 	if err != nil {
 		return nil, fmt.Errorf("couldn't create new user: %w", err)
 	}
 	err = db.Bun().NewSelect().Table("users").Where("id = ?", uid).Scan(context.TODO(), &user)
 	return &user, err
 }
+
+func TestSetActive(t *testing.T) {
+	var testUsers []model.UserID
+	for _, status := range []bool{true, false} {
+		u, err := addTestUser(nil, func(u *model.User) { u.Active = status })
+		require.NoError(t, err)
+		testUsers = append(testUsers, u.ID)
+	}
+
+	type args struct {
+		updateIDs []model.UserID
+		status    bool
+	}
+	tests := []struct {
+		name string
+		args args
+	}{
+		{
+			name: "empty ok",
+			args: args{
+				updateIDs: []model.UserID{},
+				status:    false,
+			},
+		},
+		{
+			name: "set assorted to active",
+			args: args{
+				updateIDs: testUsers,
+				status:    true,
+			},
+		},
+		{
+			name: "set back to inactive assorted to active",
+			args: args{
+				updateIDs: testUsers,
+				status:    false,
+			},
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			err := SetActive(context.Background(), tt.args.updateIDs, tt.args.status)
+			require.NoError(t, err)
+
+			if len(tt.args.updateIDs) == 0 {
+				return
+			}
+
+			for _, id := range testUsers {
+				fu, err := ByID(context.Background(), id)
+				require.NoError(t, err)
+				require.Equal(t, fu.Active, tt.args.status)
+			}
+		})
+	}
+}
+
+func TestProfileImage(t *testing.T) {
+	u, err := addTestUser(nil)
+	require.NoError(t, err)
+
+	var fakeImage [16]byte
+	_, err = rand.Read(fakeImage[:])
+	require.NoError(t, err)
+
+	_, err = db.Bun().NewInsert().Model(&UserProfileImage{
+		UserID:   u.ID,
+		FileData: fakeImage[:],
+	}).Exec(context.Background())
+	require.NoError(t, err)
+
+	resImage, err := ProfileImage(context.Background(), u.Username)
+	require.NoError(t, err)
+	require.Equal(t, fakeImage[:], resImage, "received image wasn't correct")
+}
+
+func TestUpdateUserSettings(t *testing.T) {
+	u, err := addTestUser(nil)
+	require.NoError(t, err)
+
+	t.Log("noop reset is fine")
```

Comment:
thoughts on the logs? I don't really love them since I tend to run tests with ```-v```

Reply:
I like them when the test fails, usually. For a longer test like this.

Reply:
I can just make them comments though.